axelrod/tests/unit/test_strategy_transformers.py

```diff
@@ -101,38 +248,60 @@ def test_dual_wsls_transformer(self):
         for _ in range(10):
             p2.play(p3)
 
-        self.assertEqual(p1.history, [x.flip() for x in p2.history])
+        flip_play_attributes(p1)
+        self.assertEqual(p1.history, p2.history)
+        self.assertEqual(p1.cooperations, p2.cooperations)
+        self.assertEqual(p1.defections, p2.defections)
+        self.assertEqual(p1.state_distribution, p2.state_distribution)
 
-    def test_dual_tft_transformer(self):
+    def test_dual_transformer_with_all_strategies(self):
         """Tests that DualTransformer produces the opposite results when faced
         with the same opponent history.
         """
-        p1 = axelrod.TitForTat()
-        p2 = DualTransformer()(axelrod.TitForTat)()
-        p3 = axelrod.CyclerCCD()  # Cycles 'CCD'
+        for s in axelrod.strategies:
+            self.assert_dual_wrapper_correct(s)
 
-        for _ in range(10):
-            p1.play(p3)
+    def test_dual_jossann_regression_test(self):
+        klass = JossAnnTransformer((0.2, 0.3))(axelrod.Alternator)
+        self.assert_dual_wrapper_correct(klass)
 
-        p3.reset()
-        for _ in range(10):
-            p2.play(p3)
+        klass = JossAnnTransformer((0.5, 0.4))(axelrod.EvolvedLookerUp2_2_2)
+        self.assert_dual_wrapper_correct(klass)
 
-        self.assertEqual(p1.history, [x.flip() for x in p2.history])
+    def test_dual_transformer_special_cases_1(self):
```

Comment:
yes, that is correct.  it addresses issues with the original code and a few different tries on the re-write